--- Objects/fileobject.c
+++ Objects/fileobject.c
@@ -519,7 +519,7 @@
 */

 static PyObject *
-getline(f, n)
+mygetline(f, n)
 	PyFileObject *f;
 	int n;
 {
@@ -641,7 +641,7 @@
 	}
 	if (((PyFileObject*)f)->f_fp == NULL)
 		return err_closed();
-	return getline((PyFileObject *)f, n);
+	return mygetline((PyFileObject *)f, n);
 }

 /* Python method */
@@ -661,7 +661,7 @@
 		return PyString_FromString("");
 	if (n < 0)
 		n = 0;
-	return getline(f, n);
+	return mygetline(f, n);
 }

 static PyObject *
@@ -755,7 +755,7 @@
 			goto error;
 		if (sizehint > 0) {
 			/* Need to complete the last line */
-			PyObject *rest = getline(f, 0);
+			PyObject *rest = mygetline(f, 0);
 			if (rest == NULL) {
 				Py_DECREF(line);
 				goto error;
