--- Parser/asdl_c.py
+++ Parser/asdl_c.py
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2
 """Generate C code from an ASDL description."""

 # TO DO
@@ -990,7 +990,7 @@
             self.emit("case %s:" % t.name, 2)
             self.emit("Py_INCREF(%s_singleton);" % t.name, 3)
             self.emit("return %s_singleton;" % t.name, 3)
-        self.emit("default:" % name, 2)
+        self.emit("default:", 2)
         self.emit('/* should never happen, but just in case ... */', 3)
         code = "PyErr_Format(PyExc_SystemError, \"unknown %s found\");" % name
         self.emit(code, 3, reflow=False)
--- Parser/asdl.py
+++ Parser/asdl.py
@@ -16,8 +16,8 @@

 import spark

-def output(string):
-    sys.stdout.write(string + "\n")
+def output(*args):
+    sys.stdout.write(" ".join(str(a) for a in args) + "\n")


 class Token(object):
